exchange/src/wrapper/algo_wrapper/binary/binary_wrapper.cpp

```diff
@@ -0,0 +1,143 @@
+#include "binary_wrapper.hpp"
+
+#include "common/util.hpp"
+#include "wrapper/messaging/exchange_communicator.hpp"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/select.h>
+#include <unistd.h>
+
+#include <cerrno>
+#include <cstdlib>
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <stdexcept>
+
+namespace {
+namespace fs = std::filesystem;
+
+std::pair<int, std::filesystem::path>
+get_temp_file()
+{
+#ifdef __APPLE__
+    std::string template_path = (fs::temp_directory_path() / "algoXXXXXX").string();
+    std::vector<char> writable_template_path(
+        template_path.begin(), template_path.end()
+    );
+    writable_template_path.push_back('\0');
+    int fd = mkstemp(writable_template_path.data());
+    if (fd == -1) {
+        throw std::runtime_error("Failed to get file descriptor for temporary file");
+    }
+
+    return {fd, writable_template_path.data()};
+
+#else
+    int memfd = memfd_create("algo", MFD_CLOEXEC);
+    if (memfd == -1) {
+        throw std::runtime_error("Failed to create memfd");
+    }
+
+    return {memfd, "/proc/self/fd/" + std::to_string(memfd)};
+
+#endif
+}
+
+} //   namespace
+
+namespace nutc::wrapper {
+
+BinaryWrapper::BinaryWrapper(
+    std::string algo, std::string trader_id, ExchangeCommunicator communicator
+) : Wrapper(std::move(algo), std::move(trader_id), std::move(communicator))
+{
+    auto [fd, path] = get_temp_file();
+
+    fd_ = fd;
```

Comment:
No. On a small scale u can use std::tie, on a larger scale i think this assignment could just be restructured